src/workerd/util/sqlite.c++

```diff
@@ -236,6 +236,37 @@ static constexpr kj::StringPtr ALLOWED_SQLITE_FUNCTIONS[] = {
   "json_tree"_kj,
 };
 
+// https://www.sqlite.org/pragma.html
+static constexpr kj::StringPtr ALLOWED_READ_PRAGMAS[] = {
+  // We allowlist these SQLite pragmas (for read only, never with arguments).
+  "data_version"_kj,
+  "page_count"_kj,
+  "freelist_count"_kj,
+
+  // Let a user see some internal stats?
+  "max_page_count"_kj,
+  "page_size"_kj,
```

Comment:
I guess the reason to expose this is because otherwise page_count is not meaningful?

I am nervous, though, as this feels like an internal implementation detail.

Another option here would be to provide a JavaScript API to get database stats, where we could then return bytes. I'm more comfortable with this since we can control it more easily. We have zero control over the workings of these pragmas.

But unless there is an immediate need I would prefer to punt all this for now.

Reply:
Yeah the immediate need is D1's dashboard queries, but I see the code in #619 gives us a way to do that.